src/identity/hooks/useName.ts

```diff
@@ -9,29 +9,42 @@ import type { Address, GetEnsNameReturnType } from 'viem';
  * @param address - The Ethereum address for which the ENS name is being fetched.
  * @returns A promise that resolves to the ENS name (as a string) or null.
  */
-export const ensNameAction = (address: Address) => async (): Promise<GetEnsNameReturnType> => {
-  try {
-    return await publicClient.getEnsName({
-      address,
-    });
-  } catch (err) {
-    return null;
-  }
+export const ensNameAction = async (address: Address): Promise<GetEnsNameReturnType> => {
+  return await publicClient.getEnsName({
+    address,
+  });
+};
+
+type Arguments = {
```

Comment:
what's our thinking around make this an object?

Like beside `address` what do we want more in the future as arguments?

Reply:
I always advocate for a convention where if there's more than 1 argument, to always use object. Here's Cory echoing this too

https://twitter.com/housecor/status/1763604283662999819

Reply:
Although actually in this case I made 2 objects, which might come off as opiniated pattern to separate tanstack query arguments

Reply:
I think we need to create an API that allows easily to switch between ENS, Nouns, Lens, etc.. so extra arguments are probalby need it.

Reply:
yeah I think something like that would be good perhaps in future

`useName({address, providers: {ens: true, nouns: true, lens: true}})`